### PR TITLE
fix(adepter-pg): normalize MONEY values for Decimal

### LINKED ISSUE #28674: [Prisma 7.0.0] [DecimalError] Invalid argument
### Bug description

Hello, i've got an error while migrating prisma to latest version with the `money` type as a column type:

```
[DecimalError] Invalid argument: 50,000.00
```

in the previous version 6.x.x it worked well, because we're not uses adapter, i've seen in the `@prisma/adapter-pg` there is a function where they normalize the money, but it's only `slice(1)`:

Image

i've tried overrides the parser like this:
```typescript
const pool = new pg.Pool({ connectionString: process.env.DB_URL });
const adapter = new PrismaPg(pool, {
  async userDefinedTypeParser(oid, value) {
    return pg.types.getTypeParser(oid)(value);
  },
  schema: 'public',
});
pg.types.setTypeParser(pg.types.builtins.MONEY, (val) => {
  if (val === null) return null;
  const clean = val.replace(/[^0-9.-]+/g, '');
  return clean;
});
```

but still it won't work

my query

```typescript
    const monthlyRevenue = this.prisma.transactionOrders.aggregate({
      _sum: { grandTotal: true },
      where: {
        createdAt: {
          gte: firstDayOfThisMonth,
          lte: lastDayOfThisMonth,
        },
        status: OrderStatus.PAID,
      },
    });
```

### Severity

⚠️ Major: Breaks core functionality (e.g., migrations fail)

### Reproduction

just do the query with column that has type `Money`

```typescript
    const monthlyRevenue = this.prisma.transactionOrders.aggregate({
      _sum: { grandTotal: true },
      where: {
        createdAt: {
          gte: firstDayOfThisMonth,
          lte: lastDayOfThisMonth,
        },
        status: OrderStatus.PAID,
      },
    });
```

### Expected vs. Actual Behavior

Expect: the decimal should return value
Actual: `[DecimalError] Invalid argument: 50,000.00`

### Frequency

Consistently reproducible

### Does this occur in development or production?

Both development and production

### Is this a regression?

Yes, it is worked in the prisma version 6

### Prisma Schema & Queries

```prisma
generator client {
  provider     = "prisma-client"
  output       = "../src/prisma"
  moduleFormat = "cjs"
}

datasource db {
  provider = "postgresql"
}

```
```ts
    const monthlyRevenue = this.prisma.transactionOrders.aggregate({
      _sum: { grandTotal: true },
      where: {
        createdAt: {
          gte: firstDayOfThisMonth,
          lte: lastDayOfThisMonth,
        },
        status: OrderStatus.PAID,
      },
    });
```

### Prisma Config

```ts
import 'dotenv/config';
import { defineConfig, env } from 'prisma/config';

export default defineConfig({
  schema: 'prisma',
  datasource: {
    url: env('DB_URL'),
  },
});

```

### Logs & Debug Info

```
{
  url: '/api/v1/administrator/home/event-stats',
  method: 'GET',
  msg: '[DecimalError] Invalid argument: 50,000.00',
  stack: 'Error: [DecimalError] Invalid argument: 50,000.00\n    at parseOther (/Users/rashgaroth/Documents/NvProductions/nvp-na-main-service/node_modules/@prisma/client-runtime-utils/dist/index.js:1831:11)\n    at new Decimal2 (/Users/rashgaroth/Documents/NvProductions/nvp-na-main-service/node_modules/@prisma/client-runtime-utils/dist/index.js:2213:55)\n    at deserializeTaggedValue (/Users/rashgaroth/Documents/NvProductions/nvp-na-main-service/node_modules/@prisma/client-engine-runtime/src/json-protocol.ts:152:14)\n    at mapper (/Users/rashgaroth/Documents/NvProductions/nvp-na-main-service/node_modules/@prisma/client-engine-runtime/src/json-protocol.ts:127:14)\n    at mapObjectValues (/Users/rashgaroth/Documents/NvProductions/nvp-na-main-service/node_modules/@prisma/client-engine-runtime/src/json-protocol.ts:110:19)\n    at le (/Users/rashgaroth/Documents/NvProductions/nvp-na-main-service/node_modules/@prisma/client-engine-runtime/src/json-protocol.ts:135:12)\n    at Array.map (<anonymous>)\n    at mapper (/Users/rashgaroth/Documents/NvProductions/nvp-na-main-service/node_modules/@prisma/client-engine-runtime/src/json-protocol.ts:122:19)\n    at mapObjectValues (/Users/rashgaroth/Documents/NvProductions/nvp-na-main-service/node_modules/@prisma/client-engine-runtime/src/json-protocol.ts:110:19)\n    at le (/Users/rashgaroth/Documents/NvProductions/nvp-na-main-service/node_modules/@prisma/client-engine-runtime/src/json-protocol.ts:135:12)'
}
```

### Environment & Setup

- **OS:** 
- **Database:** 
- **Node.js version:** 

### Prisma Version

```
7.0.0
```

### PR DESCRIPTION
**Pull request by SHUBHI-KUMARI**
## Summary

This PR fixes the parsing of PostgreSQL `MONEY` values in `@prisma/adapter-pg`.  
Currently, `normalize_money` simply uses `money.slice(1)`, which only removes the
leading currency symbol and does not handle thousands separators, locale formats,
or values without a symbol. This causes Prisma to pass invalid strings such as
`"50,000.00"` to `Decimal`, resulting in:

`[DecimalError] Invalid argument: 50,000.00`

This behavior worked in Prisma 6 and regressed in Prisma 7.

## Changes

- Updated `normalize_money` to:
  - Strip currency symbols, spaces, and letters
  - Remove thousands separators
  - Detect locale decimal separators
  - Normalize all MONEY strings into valid `Decimal`-compatible numeric strings
- Supports both US-style (`$50,000.00`) and EU-style (`€1.234,56`) formats.

## Reproduction

Using PostgreSQL with a `MONEY` column:

```prisma
model Order {
  id        Int      @id @default(autoincrement())
  price     Decimal  @db.Money
  createdAt DateTime @default(now())
}
````

Aggregating:

```ts
await prisma.order.aggregate({
  _sum: { price: true }
});
```

### Before (Prisma 7.0.0)

Throws:
`[DecimalError] Invalid argument: 50,000.00`

### After (this PR)

Returns the correct aggregated decimal value with no errors.
closes: #28674

## Regression

This issue did not exist in Prisma 6.x.
The regression appears after introduction of the new adapter conversion pipeline in Prisma 7.

## Testing

Verified locally using the PostgreSQL sandbox with MONEY values inserted such as:

* `$50,000.00`
* `€1.234,56`

Aggregate queries now succeed without errors and return correct numeric results.
closes: #28674

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved PostgreSQL money type handling to correctly parse US/EU formats, currency symbols, negative representations (including parentheses), ambiguous separators, leading zeros and other edge cases.

* **Tests**
  * Added extensive unit and functional tests covering normalization, parsing, aggregations, raw queries and many edge-case scenarios.

* **Documentation**
  * Added a manual test guide detailing end-to-end verification steps for money-type behavior.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by jacek-prisma — CHANGES_REQUESTED** [review]
Thank you for the contribution. You will need to add tests that cover the functionality implemented here. The easiest way is to use one of the functional tests like `packages/client/tests/functional/unixepoch-ms-datetime` as a template.

**Comment by SHUBHI-KUMARI** [comment]
Hi! @jacek-prisma
I've added functional tests as requested. The tests are located in `/packages/client/tests/functional/pg-money-type/` and follow the same pattern as `unixepoch-ms-datetime`.

## Test Structure

- `_matrix.ts` - Configures PostgreSQL as the test provider
- `prisma/_schema.ts` - Defines an Order model with a `Decimal @db.Money` field
- `tests.ts` - Contains 22 test cases covering:
  - US format (comma thousands, period decimal)
  - EU format (period thousands, comma decimal)
  - Negative values (minus sign and parentheses notation)
  - Edge cases (zeros, large numbers, leading zeros)
  - **Aggregations (SUM, MIN, MAX, COUNT)** - the original bug scenario
  - Ambiguous separator disambiguation
  - Malformed input handling
  - Raw queries with formatted values

The tests verify that formatted money strings like `$50,000.00` work correctly with aggregations without throwing `DecimalError`, which was the core issue reported.

## Test Execution

The functional tests would run once the test database infrastructure is configured. In the meantime, all 88 unit tests in `@prisma/adapter-pg` are passing, including the 54 comprehensive tests specifically for the `normalize_money` function that verify the fix handles all edge cases correctly.

Let me know if you'd like any adjustments to the test structure or coverage!

**Comment by SHUBHI-KUMARI** [comment]
hey @jacek-prisma, you haven't given a review on this yet. I am waiting for it.

File: packages/adapter-pg/src/conversion.ts [hunk 1]
```diff
@@ -332,7 +332,86 @@ function normalize_timez(time: string): string {
...
+    cleaned = cleaned.replace(/^0+(?=\d)/, '')
+    if (cleaned.startsWith('.')) {
+      cleaned = '0' + cleaned
+    }
+  } else {
+    cleaned = cleaned.replace(/^0+/, '') || '0'
+  }
+
+  return minusSign + cleaned
 }
```
[the remark this answers is not in the document]
Comment (SHUBHI-KUMARI) [reply packages/adapter-pg/src/conversion.ts:415]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/prisma/prisma/pull/28681#discussion_r2563494898)

**Comment by jacek-prisma** [comment]
The underlying issue has already been addressed in https://github.com/prisma/prisma-engines/pull/5705 in a slightly different, but more general way. You should be able to try it out with a `dev` release soon: https://www.npmjs.com/package/prisma?activeTab=versions (it's not available yet).

**Comment by SamuelMS** [comment]
> The underlying issue has already been addressed in [prisma/prisma-engines#5705](https://github.com/prisma/prisma-engines/pull/5705) in a slightly different, but more general way. You should be able to try it out with a `dev` release soon: https://www.npmjs.com/package/prisma?activeTab=versions (it's not available yet).

@jacek-prisma – I believe this fix doesn't work for sum / avg queries. Would it be possible to extend it to support money fields for those as well (which presently crash otherwise in 7.8.0)?

**Comment by tensordreams** [comment]
Hi @SHUBHI-KUMARI — thanks for this, and apologies for the slow reply.

We're clearing the open PR backlog ahead of moving this codebase to a `v7` branch, with the [Prisma Next](https://github.com/prisma/prisma-next) rewrite taking over `main`.

We are closing this one: the underlying MONEY handling issue was addressed more generally in [prisma-engines#5705](https://github.com/prisma/prisma-engines/pull/5705), which has merged. Keeping a second adapter-side normalization path would duplicate that fix.

Thanks for the time you put into it.
